lib/service/connect.go

```diff
@@ -684,6 +687,108 @@ func (process *TeleportProcess) firstTimeConnect(role types.SystemRole) (*Connec
 	return connector, nil
 }
 
+func (process *TeleportProcess) initOpenSSH() {
+	process.RegisterWithAuthServer(types.RoleNode, SSHIdentityEvent)
+	process.SSHD = openssh.NewSSHD(process.Config.OpenSSH.RestartCommand)
+	process.RegisterCriticalFunc("common.rotate_openssh2", process.syncOpenSSHRotationState)
+}
+
+func (process *TeleportProcess) syncOpenSSHRotationState() error {
+	if _, err := process.WaitForEvent(process.GracefulExitContext(), TeleportReadyEvent); err != nil {
+		return trace.Wrap(err)
+	}
+	conn, err := process.WaitForConnector(SSHIdentityEvent, nil)
+	if conn == nil {
+		return trace.Wrap(err)
+	}
+	defer conn.Close()
+
+	status, err := process.syncRotationState(conn)
```

Comment:
I notice that this is otherwise called from `syncRotationStateAndBroadcast`, can it ever be a problem that we sync but don't "broadcast" here?

Reply:
The only relevant thing that `syncRotationStateAndBroadcast` does is broadcast a `TeleportReloadEvent` if the host credentials have changed and Teleport should restart, but we are doing this one-shot thing, so we shouldn't actually reload.